File: minimum positive and maximum negative in array/minPositive_maxNegative.js

var arr = Array.from({length: 100}, () => Math.floor(Math.random() * 200) - 100);

var minP = 2147483648;
var maxN = -2147483648;

for (var i = 0; i < arr.length; i++) {
    if(arr[i] >= 0){
        if(arr[i] < minP){
            minP = arr[i];
        }
    } else {
        if(arr[i] > maxN){
            maxN = arr[i];
        }
    }
}

console.log("minP: " + minP + " maxN: " + maxN);

File: minimum positive and maximum negative in array/minPositive_maxNegative.cpp
#include <iostream>
#include <ctime>

using namespace std;

int main() {
    srand(time(0));
    int arr[20];
    for (int i = 0; i < 20; ++i) {
        arr[i] = rand() % 201 - 100;
    }

    int minP = 0x7FFFFFFF;
    int maxN = 0x80000000;
    for (int i = 0; i < sizeof(arr)/sizeof(arr[0]); ++i) {
        if(arr[i] > 0){

        if(arr[i] < minP){
            minP = arr[i];
        }
    } else {
        if(arr[i] > maxN){
            maxN = arr[i];
        }
    }
}
    cout << "Min Positive: " << minP << ", Max Negative: " << maxN << endl;

    return 0;
}